### DISCUSSION TITLE
Delayed typing

### DISCUSSION THREAD

**Discussion by ace540i** [krisk/Fuse#540] [Q&A] [answered]
Hi, i've been using fuse and it's a good product, thank you!
i'm using with react and using state and useEffect to update table in my screen. 

when i type into the search box it is very slow.  table is 7 columns, 23 rows 1.7kb. 
all columns are in filter keys. 

not sure if it's react or fuse that is slow.

**Comment by krisk** [comment, marked as answer]
Very late reply — sorry about that.

23 rows at 1.7kb is tiny for Fuse, so the slowness is almost certainly coming from React re-renders, not the search itself. I'd recommend debouncing the input (e.g. 150-200ms) so Fuse isn't invoked on every keystroke, and making sure you're not re-creating the Fuse instance on each render.
